test/engine/SpatialJoinTest.cpp

```diff
@@ -0,0 +1,1957 @@
+// is a row of this column))
+std::vector<std::vector<std::string>> orderColAccordingToVarColMap(
+    VariableToColumnMap varColMaps,
+    std::vector<std::vector<std::string>> columns,
+    std::vector<std::string> columnNames) {
+  std::vector<std::vector<std::string>> result;
+  auto indVariableMap = copySortedByColumnIndex(varColMaps);
+  for (size_t i = 0; i < indVariableMap.size(); i++) {
+    for (size_t k = 0; k < columnNames.size(); k++) {
+      if (indVariableMap.at(i).first.name() == columnNames.at(k)) {
+        result.push_back(columns.at(k));
+        break;
+      }
+    }
+  }
+  return result;
+}
+
+// helper function to create a vector of strings representing rows, from a
+// vector of strings representing columns. Please make sure, that the order of
+// the columns is already matching the order of the result. If this is not the
+// case call the function orderColAccordingToVarColMap
+std::vector<std::string> createRowVectorFromColumnVector(
+    std::vector<std::vector<std::string>> column_vector) {
+  std::vector<std::string> result;
+  if (column_vector.size() > 0) {
+    for (size_t i = 0; i < column_vector.at(0).size(); i++) {
+      std::string str = "";
+      for (size_t k = 0; k < column_vector.size(); k++) {
+        str += column_vector.at(k).at(i);
+        str += " ";
+      }
+      result.push_back(str.substr(0, str.size() - 1));
+    }
+  }
+  return result;
+}
+
+// this function compares the ResultTable resultTableToTest with the
+// expected_output. It assumes, that all rows are unique. Therefore the tables
+// are equal, if each row of the expected_output is contained in the table and
+// the tables have the same size
+void compareResultTable(  // const QueryExecutionContext* qec,
+                          // const std::shared_ptr<const ResultTable>
+                          // resultTableToTest,
+    std::vector<std::string> tableToTest,
+    std::vector<std::string>* expected_output) {
+  // rows will be a reordered version of the tableToTest
+  std::vector<std::string> rows(expected_output->size(), "");
+  // std::vector<std::string> tableToTest = printTable(qec, resultTableToTest);
+  for (size_t i = 0; i < expected_output->size(); i++) {
+    rows.at(i) = tableToTest.at(i);  // initialize as copy, reorder when needed
+  }
+  for (size_t i = 0; i < expected_output->size(); i++) {
+    for (size_t k = 0; k < tableToTest.size(); k++) {
+      if (tableToTest.at(k) == expected_output->at(i)) {
+        rows.at(i) = tableToTest.at(k);  // change order of expected output
+        break;
+      }
+    }
+  }
+  // if all rows from the expected output are in the result table and the tables
+  // are equally large, then both tables are equal (assuming each row is unique)
+  ASSERT_EQ(tableToTest.size(), expected_output->size());
+  for (size_t i = 0; i < expected_output->size(); i++) {
+    ASSERT_EQ(expected_output->at(i), rows.at(i));
+  }
```

Comment:
This looks like a rather clunky implementation of a simple
` EXPECT_THAT(tableToTest, ::testing::UnorderedElementsAreArray(*expected_output))`

(GTest is your friend).

Reply:
(you need to include `<gmock/gmock.h>` for this.